contracts/src/lido/L1LidoGateway.sol

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity =0.8.16;
+
+import {IL1ERC20Gateway} from "../L1/gateways/IL1ERC20Gateway.sol";
+import {L1ERC20Gateway} from "../L1/gateways/L1ERC20Gateway.sol";
+import {IL1ScrollMessenger} from "../L1/IL1ScrollMessenger.sol";
+import {IL2ERC20Gateway} from "../L2/gateways/IL2ERC20Gateway.sol";
+import {ScrollGatewayBase} from "../libraries/gateway/ScrollGatewayBase.sol";
+
+import {LidoBridgeableTokens} from "./LidoBridgeableTokens.sol";
+import {LidoGatewayManager} from "./LidoGatewayManager.sol";
+
+contract L1LidoGateway is L1ERC20Gateway, LidoBridgeableTokens, LidoGatewayManager {
+    /**********
+     * Errors *
+     **********/
+
+    /// @dev Thrown when deposit zero amount token.
+    error ErrorDepositZeroAmount();
+
+    /*************
+     * Variables *
+     *************/
+
+    /// @dev The initial version of `L1LidoGateway` use `L1CustomERC20Gateway`. We keep the storage
+    /// slot for `tokenMapping` for compatibility. It should no longer be used.
+    mapping(address => address) private __tokenMapping;
+
+    /***************
+     * Constructor *
+     ***************/
+
+    /// @param _l1Token The address of the bridged token in the L1 chain
+    /// @param _l2Token The address of the token minted on the L2 chain when token bridged
+    constructor(address _l1Token, address _l2Token) LidoBridgeableTokens(_l1Token, _l2Token) {
+        _disableInitializers();
+    }
+
+    /// @notice Initialize the storage of L1LidoGateway v1.
+    function initialize(
+        address _counterpart,
+        address _router,
+        address _messenger
+    ) external initializer {
+        require(_router != address(0), "zero router address");
+
+        ScrollGatewayBase._initialize(_counterpart, _router, _messenger);
+    }
+
+    /// @notice Initialize the storage of L1LidoGateway v2.
+    function initializeV2() external reinitializer(2) {
+        __LidoGatewayManager_init();
+    }
+
+    /*************************
+     * Public View Functions *
+     *************************/
+
+    /// @inheritdoc IL1ERC20Gateway
+    function getL2ERC20Address(address _l1Token)
```

Comment:
Do we want to add `getL1ERC20Address`?

Reply:
I don't think we need this. The `getL2ERC20Address` is inherited function from `IL1ERC20Gateway`. If we decide to add `getL1ERC20Address`, it should be added to `IL1ERC20Gateway`, and will change a lot in the gateway contracts.